compiler/sctracker.cpp

```diff
@@ -384,15 +383,22 @@ stack_scope_id()
   return sStackScopes.back().scope_id;
 }
 
+methodmap_t::methodmap_t(methodmap_t* parent, LayoutSpec spec, const char* name)
+ : parent(parent),
+   tag(0),
+   nullable(false),
+   keyword_nullable(false),
+   spec(spec),
+   dtor(nullptr),
+   ctor(nullptr)
+{
+  ke::SafeStrcpy(this->name, sizeof(this->name), name);

```

Comment:
any reason to keep `name` in `methodmap_t` a static sized char array as opposed to just a `ke::AString`? Then you could just use the constructor in the initializer list while removing a bad `sNAMEMAX`

Reply:
I dislike AString enough that I'm currently debating whether to go whole-hog and move spcomp to std::string (and later vector). Since it's a standalone executable there should be no problems.

But, it's a change in semantics, since sNAMEMAX is embedded everywhere. I'd want to open that can of worms in a separate patch.

Reply:
That makes sense, and adding std::string is probably a good move as well. 👍